src/main/java/seedu/contax/logic/parser/RangeEditCommandParser.java

```diff
@@ -0,0 +1,88 @@
+package seedu.contax.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.contax.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_RANGE_FROM;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_RANGE_TO;
+import static seedu.contax.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.contax.commons.core.index.Index;
+import seedu.contax.logic.commands.EditCommand;
+import seedu.contax.logic.commands.RangeEditCommand;
+import seedu.contax.logic.parser.exceptions.ParseException;
+import seedu.contax.model.tag.Tag;
+
+
+
+public class RangeEditCommandParser {
+    /**
+     * Parses the given {@code String} of arguments in the context of the RangeEditCommandParser
+     * and returns an RangeEditCommandParser object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public RangeEditCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_RANGE_FROM, PREFIX_RANGE_TO,
+                        PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+
+        Index fromIndex;
+        Index toIndex;
+        EditCommand.EditPersonDescriptor editPersonDescriptor = new EditCommand.EditPersonDescriptor();
+        if (argMultimap.getValue(PREFIX_RANGE_FROM).isPresent()) {
+            fromIndex = Index.fromOneBased(Integer.parseInt(argMultimap.getValue(PREFIX_RANGE_FROM).get()));
+        } else {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
+        }
+
+        if (argMultimap.getValue(PREFIX_RANGE_TO).isPresent()) {
```

Comment:
I'm not sure if this is a good idea but should we consider extracting this function to something like `argMultiMap#isPrefixPresent(Prefix)` to improve readability? @sharpstorm can probably weigh in on this.

Reply:
Yes, it is probably a good idea, but lets finish up functionality for 1.2b before attempting to refactor.

Reply:
since different prefix will have different set method and some different behavior like throw exceptions, i think maybe not easy to extract